lexer/lexer.go

```diff
@@ -340,10 +337,41 @@ func (l *Lexer) prevChar(steps int) byte {
 
 func (l *Lexer) readIdentifier() string {
 	position := l.position
-	for isLetter(l.ch) {
+	for isIdentifier(l.ch) {
 		l.readChar()
 	}
-	return l.input[position:l.position]
+	return string(l.input[position:l.position])
+}
+
+func isIdentifier(ch rune) bool {
+	return !isSpecialRune(ch) && !isDigit(ch) && !isEmpty(ch)
```

Comment:
I think we should change this as it's risky. I would suggest keeping the isLetter function and convert it from what it currently is to checking unicode letters rather than just ascii ones -- we could use `unicode.IsLetter(...)` instead of checking a z like we were doing before this PR.

Reply:
Thanks for these bomb comments!! Really helped to clarify what I was trying to do 😆